utils/mochaRunner/src/main.ts

```diff
@@ -0,0 +1,190 @@
+      process.exit(1);
+    }
+
+    if (!testSuite.platforms.includes(platform)) {
+      console.warn(
+        `Test suite ${testSuiteId} is not enabled for your platform. Running it anyway.`
+      );
+    }
+
+    applicableSuites = [testSuite];
+  }
+
+  console.log('Planning to run the following test suites', applicableSuites);
+
+  let fail = false;
+  const recommendations = [];
+  try {
+    for (const suite of applicableSuites) {
+      const parameters = suite.parameters;
+
+      const applicableExpectations = filterByParameters(
+        filterByPlatform(expectations, platform),
+        parameters
+      );
+
+      const skippedTests = getSkippedTests(applicableExpectations);
+
+      const env = extendProcessEnv([
+        ...parameters.map(param => {
+          return parsedSuitesFile.parameterDefinitons[param];
+        }),
+        {
+          PUPPETEER_SKIPPED_TEST_CONFIG: JSON.stringify(
+            skippedTests.map(ex => {
+              return ex.testIdPattern;
+            })
+          ),
+        },
+      ]);
+
+      const tmpDir = fs.mkdtempSync(
+        path.join(os.tmpdir(), 'puppeteer-test-runner-')
+      );
+      const tmpFilename = path.join(tmpDir, 'output.json');
+      console.log('Running', JSON.stringify(parameters), tmpFilename);
+      const handle = spawn(
+        'npx mocha',
+        ['--reporter=json', '--reporter-option', 'output=' + tmpFilename],
+        {
+          shell: true,
+          cwd: process.cwd(),
+          stdio: 'inherit',
+          env,
+        }
+      );
+      await new Promise<void>((resolve, reject) => {
+        handle.on('error', err => {
+          reject(err);
+        });
+        handle.on('close', () => {
+          resolve();
+        });
+      });
+      console.log('Finished', JSON.stringify(parameters));
+      try {
+        const results = readJSON(tmpFilename) as MochaResults;
+        console.log('Results from mocha');
+        console.log('Stats', JSON.stringify(results.stats));
+        results.pending.length > 0 && console.log('# Pending tests');
+        for (const test of results.pending) {
+          console.log(`\t? ${test.fullTitle} ${test.file}`);
+        }
+        results.failures.length > 0 && console.log('# Failed tests');
+        for (const test of results.failures) {
+          console.log(`\tF ${test.fullTitle} ${test.file}`, test.err);
+        }
+        const recommendation = getExpectationUpdates(
+          results,
+          applicableExpectations,
+          {
+            platforms: [os.platform()],
+            parameters,
+          }
+        );
+        if (recommendation.length > 0) {
+          fail = true;
+          recommendations.push(...recommendation);
+        } else {
+          console.log('Test run matches expecations');
+          continue;
+        }
+      } catch (err) {
+        fail = true;
+        console.error(err);
+      }
+    }
+  } catch (err) {
+    console.error(err);
```

Comment:
Maybe avoid false-positive in case of misconfiguration?
```suggestion
    fail = true;
    console.error(err);
```